core/embed/rust/src/ui/component/text/layout.rs

```diff
@@ -298,34 +324,81 @@ impl LayoutFit {
     }
 }
 
+const PREV_PAGE_ELLIPSIS: &str = "..";
+
 /// Visitor for text segment operations.
+/// Defines responses for certain kind of events encountered
+/// when processing the content.
 pub trait LayoutSink {
+    /// Text should be processed.
     fn text(&mut self, _cursor: Point, _layout: &TextLayout, _text: &str) {}
+    /// Hyphen at the end of line.
     fn hyphen(&mut self, _cursor: Point, _layout: &TextLayout) {}
+    /// Ellipsis at the end of the page.
     fn ellipsis(&mut self, _cursor: Point, _layout: &TextLayout) {}
+    /// Ellipsis at the beginning of the page.
+    fn prev_page_ellipsis(&mut self, _cursor: Point, layout: &TextLayout) -> i16 {
```

Comment:
I don't think this belongs here, sink isn't supposed to know anything about what the ellipsis looks like.

Reply:
Reworked in [cb381ea](https://github.com/trezor/trezor-firmware/pull/2803/commits/cb381ea80ecfa737360f8fb855921c98467bd163)